docs/design/marketing-offer-program.md

```diff
@@ -0,0 +1,257 @@
+  A client who has been awarded storage credit for their referrals is able to retain the awarded credits when the referral offer they received the credit under has been updated/changed
+
+### CATEGORIES OF FREE CREDIT & OFFER TYPES
+
+**Default Free Tier Offer:**
+  - This is the credit that all new users who join the network receive if they create an account without a Referral Link. Once a new user successfully creates their account they will have 14 days to use their Free Credit allowance.  
+  The Default Offer will remain constant, with no expiration date to the offer and does not have a redemption cap
+  The Offer Program Admin is able to adjust the default offer
+  One Default Offer per Tardigrade Satellite.
+  When there is a lapse of the Referral Testing Offer/the Referral Testing Offer has expired or reached a cap, the Default Offer will be the default to the Referral Testing Offer that new users will redeem until a new Referral Testing Offer is generated.
+
+**Referral Testing Offer**:
+  - This is the credit awarded to the Invitee upon creating their account using a Referral Link. This credit can have a different expiration date than the awarded Referrer Credit and the Default Free Credit.
+  An admin of the Referral Program is able to easily adjust the current offer through the  Admin Satellite GUI
+
+**Referrer Award Credit**:
+  - This is the credit awarded to the Referrer once the Invitee has created a *paid* account and has successfully paid their second invoice. This credit can have a different expiration date than the awarded Invitee Credit and the Default Free Credit.
+
+[Offer Types Table](https://docs.google.com/spreadsheets/d/1I3Do-HMNkpUpJAsebtl1NXw6-PlkoVaFKCP9e-TTgHA/edit?usp=sharing)
+
+## Design
+
+### Database
+
+**offer  table**
+```sql
+    id - int
+    name -  text
+    description - text  
+    credits - integer
+    redeemable_cap - integer
+    num_redeemed - integer
+    created_at - timestamp
+    offer_duration_days - int
+    award_credit_duration_days - int
+    invitee_credit_duration_days - int
+    type - enum[FREE_TIER, REFERRAL]
+    status - enum[ON_GOING, DEFAULT, EXPIRED, NO_STATUS]
+    PRIMARY KEY (id)
+```
+
+**user_credit table**
+```sql
+    user_id - bytea
+    offer_id - int
+    credits_earned - int
+    credits_used - int
+    credit_type - enum[AWARD, INVITEE, NO_TYPE]
+    expires_at - timestamp
+    created_at - timestamp
+    referred_by - bytea (nullable)
+    FOREIGN KEY (offer_id)
+    FOREIGN KEY (referred_by)
+    PRIMARY KEY (user_id)
+```
+
+**user table**
+```sql
+    total_Referred - int
+```
+
+### Offer Program Service
+
+**satellite/marketing/offers.go**
+- Create offers interface to interact with offer table
+
+```golang
+type DB interface {
+  ListAllOffers(ctx context.Context) ([]Offer, error)
+  GetOfferById(ctx context.Context, offerId Offer.ID) (Offer, error)
+  Update(ctx context.Context, offer *Offer) (*Offer, error)
```

Comment:
How is update handled when we get concurrent updates to the database? This implies using an transaction which would be problematic.